smartsim/launchable/job.py

```diff
@@ -50,26 +52,43 @@ def __init__(
         self,
         entity: SmartSimEntity,
         launch_settings: LaunchSettings,
+        name: str | None = None,
     ):
         super().__init__()
         self._entity = deepcopy(entity)
         self._launch_settings = deepcopy(launch_settings)
-        # TODO: self.warehouse_runner = JobWarehouseRunner
+        self._name = name if name else entity.name
+        check_name(self._name)
+
+    @property
+    def name(self) -> str:
+        """Retrieves the name of the Job."""
+        return self._name
+
+    @name.setter
+    def name(self, name: str | None) -> None:
+        """Sets the name of the Job."""
+        self._name = name if name else self._entity.name
```

Comment:
I like this check!  But do we want to silently overwwrite and not throw a warning or error?  (maybe worth a group discussion)